### ISSUE TITLE
Unusual model scores from osprey 

### ISSUE THREAD

**Issue by apeck12** [msmbuilder/msmbuilder#495]
Running osprey (with n_timescales set to 5) returned a few model scores that were negative, several that were 'nan', and one that was 8.8. Does this imply that none of the model scores from this osprey run are legitimate? Thanks.

**Comment by schwancr** [comment]
There's an issue in osprey too, see: pandegroup/osprey#48

**Comment by rmcgibbo** [comment]
Is it possible to look at these in more detail, in terms of what parameter values led to these models? I assume this is some kind of crazy overfitting, or where the train and test set don't cover at all the same portions of phase space.

This doesn't mean that the other scores from that osprey run are illegitimate, but it does mean that we should take a look at the parameters that lead to wonky behavior, because we might be able to learn something from them about particular corner cases, or ways to numerically harden one or more parts of msmbuilder.

**Comment by kyleabeauchamp** [comment]
I think I can reproduce this with some old HP35 data, with instructions here:

https://github.com/kyleabeauchamp/BadOspreyRun

The featurized dihedrals are uploaded to google drive:

https://drive.google.com/file/d/0ByBuQX6nQDxWdElmaFpHYzBMaXc/view?usp=sharing

Finally, this dataset consists of ~100 "long" trajectories from an old FAH project 10424.  It's possible that there are either some bad frames or disconnectivity in the dataset.

**Comment by kyleabeauchamp** [comment]
Some analysis of the output:  

```
import osprey.config, osprey.trials
import pandas as pd
import seaborn as sns

config = osprey.config.Config("./config.yaml")

session = config.trials()
items = [cursor.to_dict() for cursor in session.query(osprey.trials.Trial).all()]
df = pd.DataFrame(items).set_index('id')

for key in df.iloc[0].parameters.keys():
    df[key] = df.parameters.map(lambda x: x[key])


X = df.pivot_table(index="tica__n_components", columns="cluster__n_clusters", values="mean_test_score")
sns.heatmap(X, square=True, cmap='rainbow')

Y = df[["cluster__n_clusters", "tica__n_components", "mean_test_score"]].sort("mean_test_score")
Y
```

```
In [14]: df.ix[12]
Out[14]: 
completed                                             2015-03-29 15:54:53.526093
config_sha1                             efd23f1b7f3f7e65b27da4d70a55931ba77ead99
elapsed                                                           0:00:30.678719
host                                                                kyleb-DX4870
mean_test_score                                                         9.491891
mean_train_score                                                        5.896048
n_test_samples                               [65195, 65965, 60274, 63314, 60444]
n_train_samples                         [249997, 249227, 254918, 251878, 254748]
parameters                     {u'cluster__compute_labels': True, u'cluster__...
project_name                                                             default
started                                               2015-03-29 15:54:22.847374
status                                                                 SUCCEEDED
test_scores                    [5.67435242988, 5.40818458065, 5.59041940153, ...
traceback                                                                   None
train_scores                   [5.88890494375, 5.90397690991, 5.9118708304, 5...
user                                                                       kyleb
cluster__compute_labels                                                     True
cluster__max_iter                                                            100
cluster__verbose                                                               0
cluster__init                                                          k-means++
cluster__n_init                                                                3
msm__verbose                                                               False
cluster__random_state                                                       None
msm__ergodic_cutoff                                                            1
msm__n_timescales                                                              5
cluster__batch_size                                                          100
tica__weighted_transform                                                   False
cluster__max_no_improvement                                                   10
cluster__reassignment_ratio                                                 0.01
tica__gamma                                                                 0.05
msm__reversible_type                                                         mle
msm__prior_counts                                                              0
cluster__init_size                                                          None
cluster__tol                                                                   0
msm__lag_time                                                                  1
tica__n_components                                                             6
msm__sliding_window                                                         True
cluster__n_clusters                                                           19
tica__lag_time                                                                 1
Name: 12, dtype: object

In [15]: df.ix[12].parameters
Out[15]: 
{u'cluster__batch_size': 100,
 u'cluster__compute_labels': True,
 u'cluster__init': u'k-means++',
 u'cluster__init_size': None,
 u'cluster__max_iter': 100,
 u'cluster__max_no_improvement': 10,
 u'cluster__n_clusters': 19,
 u'cluster__n_init': 3,
 u'cluster__random_state': None,
 u'cluster__reassignment_ratio': 0.01,
 u'cluster__tol': 0.0,
 u'cluster__verbose': 0,
 u'msm__ergodic_cutoff': 1,
 u'msm__lag_time': 1,
 u'msm__n_timescales': 5,
 u'msm__prior_counts': 0,
 u'msm__reversible_type': u'mle',
 u'msm__sliding_window': True,
 u'msm__verbose': False,
 u'tica__gamma': 0.05,
 u'tica__lag_time': 1,
 u'tica__n_components': 6,
 u'tica__weighted_transform': False}

```

**Comment by kyleabeauchamp** [comment]
```
In [16]: df.ix[12].test_scores
Out[16]: 
[5.6743524298754995,
 5.408184580646541,
 5.590419401533443,
 5.631881991330392,
 26.0]
```

**Comment by rmcgibbo** [comment]
Matt has been looking at this as well for some small datasets. The issue is
generally related to the behavior when the train and test set state spaces
are different (eg one training set state was never visited in test set),
but beyond that i'm not too clear on the details.

The proper way to compute the score in this regime isn't exactly clear to
me.
